asv_bench/benchmarks/benchmarks.py

```diff
@@ -67,30 +67,32 @@ def execute(df):
 
 
 class BaseTimeGroupBy:
-    def setup(self, data_size, ncols=1):
-        self.df = generate_dataframe(
-            ASV_USE_IMPL, "int", data_size[1], data_size[0], RAND_LOW, RAND_HIGH
+    def setup(self, data_size, groupby_ncols=1):
+        count_cols = data_size[1] - groupby_ncols
```

Comment:
I think that logic should be located at the `generate_dataframe` function. For example:
```python
...
if groupby_ncols:
    ncols -= groupby_ncols
...
```